lib/ark_elixir/api.ex

```diff
@@ -1,19 +1,113 @@
 defmodule Ark_Elixir.Api do
     @moduledoc """
+    Operations for querying the API.
+    """
+
+    @doc """
+    Allow us to make a request on the network of our choice, using get_main or
+    get_dev, to the desired endpoint, with the possibility of passing optional
+    parameters and the matching headers.
+
+    The default network is Main.
+
+    ## Examples
 
+      iex> Ark_Elixir.Api.get("api/blocks") # Query on the main net
+      iex> Ark_Elixir.Api.get("api/blocks", [limit: 2]) # Query on the main net
+      iex> Ark_Elixir.Api.get("api/blocks", [limit: 2, network: "dev"]) # Query on the dev net
+      iex> Ark_Elixir.Api.get("api/blocks", [network: "dev", limit: 2]) # In any order
+      iex> Ark_Elixir.Api.get("api/blocks", [network: "dev"]) # No query params, just the network and the whole response
+      iex> Ark_Elixir.Api.get("api/blocks", [network: "bar"]) # Incorrect network, switching on the main one
+      iex> Ark_Elixir.Api.get("api/blocks", [network: :dev]) # You can specify the network as an Atom
     """
     def get(endpoint, opts \\ []) do
-        HTTPotion.get("https://api.arknode.net/#{endpoint}" <> query(opts),
-            headers: [nethash: "6e84d08bd299ed97c212c886c98a57e36545c8f5d645ca7eeae63a8bd62d8988", version: "1.0.1", port: "4001"])
-        |> Map.get(:body)
-        |> Poison.Parser.parse!
+      case opts[:network] do
```

Comment:
In the functions you're passing opts in as `opts \\ []` and sending them like `[address: address, network: opts]` which would set network to `[]` or whatever the user passes in.

It would make more sense to have the functions defined like this:

```
def get_accounts(address, opts \\ []) do
    Ark_Elixir.Api.get("api/accounts", [{:address, address} | opts])
end
```

This way if you ever have to refactor to add more opts they are not all placed under the "network" key.

Reply:
Oh I see, nice ! I will change that ASAP.